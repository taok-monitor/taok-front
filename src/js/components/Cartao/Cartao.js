import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardText, CardBody, CardTitle } from 'reactstrap';

const Cartao = ({ orgao, valor }) => {
    return (
        <Card>
            <CardBody>
                <CardTitle>{orgao}</CardTitle>
                <CardText>{valor}</CardText>
            </CardBody>
        </Card>
    );
};

Cartao.propTypes = {
    orgao: PropTypes.string.isRequired,
    valor: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
};

export default Cartao;
