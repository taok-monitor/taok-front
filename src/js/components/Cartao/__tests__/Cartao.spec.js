import React from 'react';
import { render } from '@testing-library/react';

import Cartao from '../Cartao';

describe('<Cartao />', () => {
    it('should render correctly with props', () => {
        const props = {
            orgao: 'Hospital Geral',
            valor: '50000',
        };

        const { container, getByText } = render(<Cartao {...props} />);

        expect(getByText(props.valor)).toBeInTheDocument();
        expect(getByText(props.orgao)).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });
});
