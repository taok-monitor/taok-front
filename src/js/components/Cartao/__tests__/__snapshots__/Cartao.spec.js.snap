// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cartao /> should render correctly with props 1`] = `
<div
  class="card"
>
  <div
    class="card-body"
  >
    <div
      class="card-title"
    >
      Hospital Geral
    </div>
    <p
      class="card-text"
    >
      50000
    </p>
  </div>
</div>
`;
